Formula/r/recc.rb

```diff
@@ -0,0 +1,196 @@
+class Recc < Formula
+  desc "Remote Execution Caching Compiler"
+  homepage "https://buildgrid.gitlab.io/recc"
+  url "https://gitlab.com/BuildGrid/buildbox/buildbox/-/archive/1.2.20/buildbox-1.2.20.tar.gz"
+  sha256 "5ad5ccbad1250accc984823bd5d78604441c7711320b0569889f45f329ed6311"
+  license "Apache-2.0"
+  head "https://gitlab.com/BuildGrid/buildbox/buildbox.git", branch: "master"
+
+  depends_on "cmake" => :build
+  depends_on "tomlplusplus" => :build
+  depends_on "abseil"
+  depends_on "c-ares"
+  depends_on "glog"
+  depends_on "grpc"
+  depends_on "openssl@3"
+  depends_on "protobuf"
+  depends_on "re2"
+  uses_from_macos "zlib"
+
+  on_macos do
+    depends_on "gflags"
+  end
+
+  on_linux do
+    depends_on "pkg-config" => :build
+    depends_on "util-linux" => :build
+  end
+
+  def install
+    buildbox_cmake_args = %W[
+      -DCASD=ON
+      -DCASD_BUILD_BENCHMARK=OFF
+      -DCASDOWNLOAD=OFF
+      -DCASUPLOAD=OFF
+      -DFUSE=OFF
+      -DLOGSTREAMRECEIVER=OFF
+      -DLOGSTREAMTAIL=OFF
+      -DOUTPUTSTREAMER=OFF
+      -DRECC=ON
+      -DREXPLORER=OFF
+      -DRUMBA=OFF
+      -DRUN_BUBBLEWRAP=OFF
+      -DRUN_HOSTTOOLS=ON
+      -DRUN_OCI=OFF
+      -DRUN_USERCHROOT=OFF
+      -DTREXE=OFF
+      -DWORKER=OFF
+      -DRECC_CONFIG_PREFIX_DIR=#{etc}/recc
+    ]
+
+    system "cmake", "-S", ".", "-B", "build", *buildbox_cmake_args, *std_cmake_args
+    system "cmake", "--build", "build", "-j", ENV.make_jobs.to_s
+    system "cmake", "--install", "build"
+
+    generate_recc_wrapper = lambda do |compiler|
+      (bin/"recc-#{compiler}").write <<~EOS
+        #!/usr/bin/env sh
+
+        #{bin}/recc $(command -v #{compiler}) "$@"
+      EOS
+    end
+
+    generate_recc_wrapper.call("cc")
+    generate_recc_wrapper.call("c++")
+    generate_recc_wrapper.call("gcc")
+    generate_recc_wrapper.call("g++")
+    generate_recc_wrapper.call("clang")
+    generate_recc_wrapper.call("clang++")
+
+    # Generate recc-server start helper script
+    (bin/"recc-server").write <<~EOS
+      #!/usr/bin/env sh
+
+      #{bin}/buildbox-casd \
+        --local-server-instance=recc-server \
+        ${1:-#{var}/recc/casd}
+    EOS
+
+    # Generate recc.conf
+    config_directory = prefix/"etc/recc/"
+    config_directory.mkpath
+    (config_directory/"recc.conf").write <<~EOS
```

Comment:
Is there no default `recc.conf` supplied in the upstream repo?

Reply:
There is no default recc.conf in upstream, as it can be used with any [REAPI](https://github.com/bazelbuild/remote-apis?tab=readme-ov-file#servers) server. It's mainly used in conjunction with a REAPI server attached to a build server. This formula aims to provide an out-of-the-box working experience, as configuring a remote server can be a daunting task for many developers